.github/workflows/sonarcloud.yml

```
@@ -0,0 +1,79 @@
+      - name: 🛠️ Build solution
+        run: dotnet build --no-restore --configuration Release
+
+      - name: ✅ Run tests and collect coverage
+        run: |
+          dotnet tool install --global dotnet-coverage
+          dotnet-coverage collect "dotnet test --no-build --configuration Release" -f xml -o "coverage.xml"
+
+      - name: 🧾 End SonarCloud analysis
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: dotnet sonarscanner end /d:sonar.token="${{ env.SONAR_TOKEN }}"
```

YAMLlint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)